src/MiniProfiler.AspNetCore/MiniProfilerMiddleware.cs

```diff
@@ -50,7 +51,8 @@ public class MiniProfilerMiddleware
             if (basePath.StartsWith("~/", StringComparison.Ordinal)) basePath = basePath.Substring(1);
             if (basePath.EndsWith("/", StringComparison.Ordinal) && basePath.Length > 2) basePath = basePath.Substring(0, basePath.Length - 1);
 
-            BasePath = new PathString(basePath);
+            MatchPath = new PathString(basePath);
+            BasePath = new PathString(basePath.EnsureTrailingSlash());
```

Comment:
I totally agree with your change, just need to make the other lib match when removing all templating...more of a note for me there.